### ISSUE TITLE
`no-duplicates` fail positive

### ISSUE THREAD

**Issue by ntnyq** [un-ts/eslint-plugin-import-x#217]
[Repro repo](https://github.com/ntnyq-dev/repro-import-x-no-duplicates)

eslint-plugin-import-x:  v4.6.1

### ESLint config

```ts
import { config, configs  } from 'typescript-eslint'
import pluginImportX from 'eslint-plugin-import-x'

const { createNodeResolver } = pluginImportX

export default config({
  extends: [configs.recommended],
  plugins: {
    'import-x': pluginImportX
  },
  settings: {
    'import-x/resolver-next': [
      createNodeResolver({
        extensions: ['.js', '.mjs', '.ts', '.mts', '.d.ts', '.json'],
      }),
    ],
  },
  rules: {
    'import-x/no-duplicates': ['error']
  }
})
```

### Code:

```ts
import { createRouter, createWebHashHistory } from 'vue-router'
import { routes } from 'vue-router/auto-routes'

export const router = createRouter({
  history: createWebHashHistory(),
  routes,
})
```

### Errors:

```
'/Users/ntnyq/Desktop/github/ntnyq/repro-import-x-no-duplicates/node_modules/.pnpm/vue-router@4.5.0_vue@3.5.13_typescript@5.7.2_/node_modules/vue-router/dist/vue-router.mjs' imported multiple times.
```

With `--fix` passed, code will change to:

```ts
import { createRouter, createWebHashHistory, routes } from 'vue-router'

export const router = createRouter({
  history: createWebHashHistory(),
  routes,
})
```

**Comment by JounQin** [comment]
<img width="700" alt="image" src="https://github.com/user-attachments/assets/64f58f34-6cf0-45e7-b925-52f7d6231ede" />

Their `import/require` conditions point to same file `./dist/vue-router.mjs`, so they are considered duplicates.

If you use [`eslint-import-resolver-typescript`](https://github.com/import-js/eslint-import-resolver-typescript), `types` condition will be preferred, then it should work as expected.

**Comment by ntnyq** [comment]
Got it. Thanks.

I'll try typescript resolver.
